modules/samples/watertank/component/api/src/main/java/org/homeautomation/watertank/api/DeviceTypeServiceImpl.java

```diff
@@ -0,0 +1,434 @@
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getAllDevices() {
+        try {
+            List<Device> userDevices =
+                    APIUtil.getDeviceManagementService().getDevicesOfUser(APIUtil.getAuthenticatedUser());
+            ArrayList<Device> userDevicesforwatertank = new ArrayList<>();
+            for (Device device : userDevices) {
+                if (device.getType().equals(DeviceTypeConstants.DEVICE_TYPE) &&
+                    device.getEnrolmentInfo().getStatus().equals(EnrolmentInfo.Status.ACTIVE)) {
+                    userDevicesforwatertank.add(device);
+                }
+            }
+            Device[] devices = userDevicesforwatertank.toArray(new Device[]{});
+            return Response.ok().entity(devices).build();
+        } catch (DeviceManagementException e) {
+            log.error(e.getErrorMessage(), e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
+        }
+    }
+
+    /**
+     * To download device type agent source code as zip file
+     *
+     * @param deviceName name for the device type instance
+     * @param sketchType folder name where device type agent was installed into server
+     * @return Agent source code as zip file
+     */
+    @Path("/device/download")
+    @GET
+    @Produces("application/zip")
+    public Response downloadSketch(@QueryParam("deviceName") String deviceName,
+                                   @QueryParam("sketchType") String sketchType) {
+        try {
+            ZipArchive zipFile = createDownloadFile(APIUtil.getAuthenticatedUser(), deviceName, sketchType);
+            Response.ResponseBuilder response = Response.ok(FileUtils.readFileToByteArray(zipFile.getZipFile()));
+            response.status(Response.Status.OK);
+            response.type("application/zip");
+            response.header("Content-Disposition", "attachment; filename=\"" + zipFile.getFileName() + "\"");
+            Response resp = response.build();
+            zipFile.getZipFile().delete();
+            return resp;
+        } catch (IllegalArgumentException ex) {
+            return Response.status(400).entity(ex.getMessage()).build();//bad request
+        } catch (DeviceManagementException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (JWTClientException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (APIManagerException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (IOException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (UserStoreException ex) {
+            log.error(ex.getMessage(), ex);
```

Comment:
repeating the same error message (i.e. e.getErrorMessage()) here, which can be found in the stack trace as well, is not a good practice.